Backend/RecipeIngredientsAlgorithm/Iteration_1/ranking.js

```diff
@@ -0,0 +1,159 @@
+
+const helper = require('./helpers')
+
+///////////////////////////////////   POINTS RULES   //////////////////////////////////////////////
+const OVERLAP_POINTS = 10
+const MEAL_TYPE_POINTS = 3
+const TIME_COOK_POINTS = 8
+const KCAL_POINTS = 8
+
+const TOP_1_2_CUISINE_POINTS = 3
+const TOP_3_4_CUISINE_POINTS = 2
+const TOP_5_CUISINE_POINTS = 1
+const RANDOM_POINTS = 10
+var maxRandomPoints = 0
+var userRandomness = 0
+
+//////////////////////////////////   POINTS    /////////////////////////////////////
+var overlapPoints = 0
+var mealTypePoints = 0
+var timeCookPoints = 0
+var idealKcalPoints = 0
+var cuisinePoints = new Map()
```

Comment:
instead of creating global vars in a file like this; lets create a class and move these to instance fields in the class. This way, we can actually enable things like parallel processing in the future.